request.go

```diff
@@ -202,18 +201,34 @@ func (req *Request) pack() (packet []byte, err error) {
 		byte(rid >> 8), byte(rid),
 	}
 
-	body, err = msgpack.Marshal(req.body)
+	enc := msgpack.NewEncoder(&h)
+	err = enc.EncodeMapLen(len(req.body))
 	if err != nil {
 		return
 	}
+	for k,v := range(req.body) {
+		err = enc.EncodeInt64(int64(k))
+		if err != nil {
+			return
+		}
+		switch vv:=v.(type) {
+		case uint32:
```

Comment:
Why do we need this special case ? 
Does uint32 encoding differs from other types ?
Is there a bug in msgpack with uint32 ?